docs: workspaces section textual and visual improvements

stefanicjuraj opened the pull request:
- added textual additions and improvements per each sub-section
- added visual outputs when executing the terminal commands with explanations
- all changes have been tested and previewed locally

Tpuljak commented:
@stefanicjuraj we have a requirement for a signoff in all commits going to daytonaio repos. Please amend a signoff with `git commit --amend -s --no-edit && git push -f`

Tpuljak commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -29,18 +39,25 @@ You can create a Workspace.
-To automatically open a new Workspace in Visual Studio Code after creation, use the `--code` flag.
-
-**Example:**
+You can open a new Workspace in Visual Studio Code after creation by using the `--code` flag.
```
`in an IDE` instead of `Visual Studio Code` since users can configure the IDE to be opened with `daytona ide`

Tpuljak commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -118,12 +176,15 @@ Refer to **[Set the Default IDE](#set-the-default-ide)** for more information.
 ## Delete Workspaces
 
-You can delete one or more Workspaces interactively.
+Daytona allows you to delete one or more Workspaces, helping you manage your development environments by removing those that are no longer needed. This ensures you maintain a clutter-free and efficient Workspace.
```
`a clutter-free and efficient Workspace` doesn't seem appropriate. I suggest removing the sentence altogether

Tpuljak commented on src/content/docs/usage/workspaces.mdx:
````diff
@@ -223,12 +328,16 @@ You can access services running in a Workspace on your local machine.
    daytona forward 4321 example-dev-env
    ```
 
+   Upon running this command, Daytona will establish a port forwarding connection from the specified port in the Workspace to your local machine or via a shareable public URL. The terminal will display a message indicating the local port where the forwarded connection is available:
````
`via a shareable public URL` the link is public only if the user supplies the `--public` flag.

Tpuljak reviewed (changes requested):
Nice work on the changes. Have a few comments below. Didn't have time to run and view this today. I'll get back to that tomorrow

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -7,20 +7,30 @@ sidebar:
 
 ### From a Configured Git Provider
 
+Creating a Workspace from a configured Git Provider allows you to link your development environment directly to a repository hosted on platforms such as GitHub, GitLab, Bitbucket, or Gitea.
```
Would slightly change this in the sense of "Creating a Workspace from a configured a Git provider like GitHub, GitLab, Bitbucket, or Gitea, allows you to simply choose from a list of your repositories when setting up a new Workspace, rather than manually entering the Git URL."

@stefanicjuraj

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -7,20 +7,30 @@ sidebar:
 
+Creating a Workspace from a configured Git Provider allows you to link your development environment directly to a repository hosted on platforms such as GitHub, GitLab, Bitbucket, or Gitea.
+
+This method ensures that your Workspace is set up with the latest codebase and configuration from the specified repository.
```
Not sure if this sentence is needed.

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -7,20 +7,30 @@ sidebar:
+
+  Targets are essential for deploying your Workspaces, so make sure you have configured at least one.
 
 ### From a Configured Git Provider
```
@stefanicjuraj not sure if we should use this title, or more specifically where do we define with out having a git provider or just selecting `Enter a custom repository URL`

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
````diff
@@ -29,18 +39,25 @@ You can create a Workspace.
+      Enter a custom repository URL
+   ```
+
+3. Enter a name for your Workspace. Provide a descriptive name for your Workspace that reflects its purpose or project.
````
@stefanicjuraj When you test daytona you will see that it creates a name for you using the info from the repo, but you can edit it to be "descriptive name for your Workspace that reflects its purpose or project"

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -29,18 +39,25 @@ You can create a Workspace.
      including setting the [Builder](../usage/builders).
    </Aside>
-4. Wait for Daytona to create the Workspace.
+4. Wait while Daytona sets up your Workspace. It handles all the initialization and configuration of your environment.
```
@stefanicjuraj i like your example above of cli output : 
```text
      Choose a Provider
      2 items
      ===
      GitHub
      Enter a custom repository URL
   ```

so doing the same here might make sense so the user knows what a correct created WS screen looks like

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -50,6 +67,8 @@ daytona create --code
 
```
@stefanicjuraj the flow migh be better if we go to OPEN WS here, then list?

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -64,42 +83,69 @@ daytona create --code
-You can open an existing Workspace in your IDE.
+Daytona allows you to open an existing Workspace in your IDE. This enables you to quickly access and work on your projects in your preferred development environment. With Daytona, you can effortlessly switch between different Workspaces and maintain an efficient workflow.
 
 **Prerequisites**
```
@stefanicjuraj are all these Prerequisites maybe overkill? perhaps just once on at the top "**A working installation of Daytona.**

  Verify your Daytona installation is up-to-date to avoid compatibility issues."

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -158,7 +236,16 @@ You can delete one or more existing Workspaces non-interactively.
+
+2. Confirm the action to delete the selected Workspace.
 
 ## IDEs
```
@stefanicjuraj can we take IDEs and put them in their own page?

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -212,8 +314,11 @@ You can access services running in a Workspace on your local machine.
 
 **Prerequisites**
 
```
Forward Ports from a Workspace title should be H2 to show up in the right meni

stefanicjuraj commented:
- incorporated creating a workspace from a git repository into one guide: through a configured git provider and entering a custom repository URL
- removed the IDEs section from workspaces, and created a new page dedicated to IDEs
- added additional cli output examples
- removed the unnecessary sentences and prerequisites, and reorganized the content to fit better

Tpuljak reviewed (approved):
Nice work!

ivan-burazin commented on src/content/docs/usage/workspaces.mdx:
```diff
@@ -64,66 +105,90 @@ daytona create --code
--   To use the Custom Image Builder, set both `--custom-image` and `--custom-image-user` flags.
--   To use the Dev Container Builder, set the`--devcontainer-path` flag to the Dev Container configuration path within the repository.
--   To explicitly specify the Builder, set the`--builder` flag.
+- To use the Custom Image Builder, set both `--custom-image` and `--custom-image-user` flags.
```
@stefanicjuraj does it make sense to remove most of this, and jsut say for more on builders go to builders (link) and show the info there?

ivan-burazin reviewed (changes requested):
Much better, I would just try to remove as many NOTES and TIPS as possible that is there for exceptions not rules. Thanks!

Looking good!

nkkko commented:
Looking great, reads well. Just few comments:
- please always think about inter-linking opportunities across docs or main page: e.g. "A working installation of Daytona." should link to installation
- we should remove copy icon from code blocks that are non functional
- re above: if you need design or style changes please create separate design issues and label them Design

stefanicjuraj commented:
- removed lengthy notes/tips regarding builders and ported them to the builders section
- added additional inter-linking across the documentation
- organized the content structure and formatting

I will open a new pull request containing the copy icon design and styling changes to fix the issue https://github.com/daytonaio/docs/issues/99.
